src/focus_manager/txm_focus_manager.js

```diff
@@ -184,23 +184,21 @@ export class TXMFocusManager {
         if (!this._isBlockingBackActions) return;
         this._isBlockingBackActions = false;
 
-        var state = history.state;
         window.removeEventListener("popstate", this.onPopState);
 
-        setTimeout(() => {
```

Comment:
Why is setTimeout no longer needed? Why is setTimeout no longer needed?

Reply:
This is the heart of the fix. By doing it later, sometimes the history "pop" causes the web page to load, before the other pop state event handlers even run

By doing it right here, the problem is avoided

In a sense I don't like the fix, because I don't completely understand it, but this is what makes things were reliably

Reply:
Ah interesting. Sounds like a tricky one to debug. Thanks for explaining